### PR TITLE
Exception handler

### PR DESCRIPTION
**Pull request by Simurana**
Exception handler
Created error handlers for RuntimeException, ConstraintViolationException, MethodArgumentNotValidException and tests for these handlers.

### CODE REVIEW

**Review by IlyaTrubchik — APPROVED** [review]

**Review by FedorDevelopmer — APPROVED** [review]

File: src/main/java/com/example/practice_modsen_shop/advice/ErrorHandlingControllerAdvice.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package com.example.practice_modsen_shop.advice;
```
Comment (egorzhurov) [inline src/main/java/com/example/practice_modsen_shop/advice/ErrorHandlingControllerAdvice.java]: rename to com.modsen.practice.exceptionhandler

File: src/main/java/com/example/practice_modsen_shop/advice/ErrorHandlingControllerAdvice.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import jakarta.validation.ConstraintViolationException;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ControllerAdvice
+public class ErrorHandlingControllerAdvice {
```
Comment (egorzhurov) [inline src/main/java/com/example/practice_modsen_shop/advice/ErrorHandlingControllerAdvice.java]: rename to ExceptionHandlerControllerAdvice

File: src/main/java/com/example/practice_modsen_shop/advice/ErrorHandlingControllerAdvice.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    @ExceptionHandler
```
Comment (egorzhurov) [inline src/main/java/com/example/practice_modsen_shop/advice/ErrorHandlingControllerAdvice.java]: @ExceptionHandler(RuntimeException)

File: src/main/java/com/example/practice_modsen_shop/advice/ErrorHandlingControllerAdvice.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        final List<Violation> violations = e.getBindingResult().getFieldErrors().stream()
+                .map(error -> new Violation(error.getField(), error.getDefaultMessage()))
+                .collect(Collectors.toList());
+        return new ValidationErrorResponse(violations);
+    }
+
+
+
+
+    @Getter
```
Comment (egorzhurov) [inline src/main/java/com/example/practice_modsen_shop/advice/ErrorHandlingControllerAdvice.java]: move to separate classes

File: src/main/java/com/example/practice_modsen_shop/advice/ErrorHandlingControllerAdvice.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    @RequiredArgsConstructor
+    public static class Violation {
+        private final String fieldName;
+        private final String message;
+    }
+
+    @Getter
+    @RequiredArgsConstructor
+    public static class ErrorResponse {
```
Comment (egorzhurov) [inline src/main/java/com/example/practice_modsen_shop/advice/ErrorHandlingControllerAdvice.java]: remove this class

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -1 +1,2 @@
 spring.application.name=practice_modsen_shop
```
Comment (egorzhurov) [inline src/main/resources/application.properties]: Shop???

File: src/test/java/com/example/practice_modsen_shop/TestController.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.ConstraintViolationException;
+import jakarta.validation.Path;
+import jakarta.validation.metadata.ConstraintDescriptor;
+
+import java.util.*;
+
+@RestController
+public class TestController {
```
Comment (egorzhurov) [inline src/test/java/com/example/practice_modsen_shop/TestController.java]: Delete this class

File: src/test/java/com/example/practice_modsen_shop/ErrorHandlingControllerAdviceTest.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+
+@WebMvcTest(TestController.class)
+@Import(ErrorHandlingControllerAdvice.class)
+public class ErrorHandlingControllerAdviceTest {
```
Comment (egorzhurov) [inline src/test/java/com/example/practice_modsen_shop/ErrorHandlingControllerAdviceTest.java]: Delete this class

**Pull request merged**